### ISSUE TITLE
ng-show, ng-hide doesn't work on ui-gmap-search-box

### ISSUE THREAD

**Issue by ajones-miovision** [angular-ui/angular-google-maps#1471]
It appears as though using ng-show or ng-hide on a ui-gmap-search-box has no affect. There are no console errors, so I assume those attributes are just being ignored by the directive?

```html
<ui-gmap-search-box
                ng-show="mapvm.isEditMode"
                template="mapvm.searchbox.template"
                events="mapvm.searchbox.events"
                position="'TOP_CENTER'">
</ui-gmap-search-box>
```

**Comment by nmccready** [comment]
Why not workaround it and wrap it in another div?

**Comment by ajones-miovision** [comment]
I tried that and it didn't work either. I'm just heading home from work. I'll try to get a plunker up tonight or first thing in the morning.

**Comment by ajones-miovision** [comment]
http://plnkr.co/edit/L3GnxC2R5wjSvLzgTbKT?p=preview

**Comment by nmccready** [comment]
Well ng-if does work a little better but yeah seems like a problem for sure.

**Comment by nmccready** [comment]
I havent dived in but first guess is $on('destroy) needs to remove it from the map. I think it is not doing that.

https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/models/parent/search-box-parent-model.coffee#L58-L59

As for ng-hide/show not sure.

@kandizzy any thoughts?

**Comment by nmccready** [comment]
http://plnkr.co/edit/FU2hLH54G2bXpGQtlF36?p=preview

**Comment by kandizzy** [comment]
@nmccready @ajones-miovision `options.visible` was [implemented in this PR](https://github.com/angular-ui/angular-google-maps/pull/912) to toggle visibility. I'll take a look  to make sure that it is still working and also look at pulling in the `ng-show` value if it exists.

**Comment by ajones-miovision** [comment]
The visibility option mostly works, but it appears as though it does not respect the initial value.
In this plunkr I have options.visible = false and yet the searchbox is still visible.

http://plnkr.co/edit/XqKQOU1JlSKLEUqRAcJp?p=preview

**Comment by nmccready** [comment]
@ajones-miovision please see #1474 and test out master

**Comment by DzmitryU** [comment]
If you'll try to set property visible to false initially, it wouldn't work correctly. Should I crate issue for this bug?

**Comment by kandizzy** [comment]
I'll take a look because I thought this was fixed in the above PR. Which version are you using?

**Comment by DzmitryU** [comment]
Currently I'm using version 2.1.0. Was this defect fixed in latest version 2.2.X?

**Comment by kandizzy** [comment]
Looks like it was fixed in 2.1.6
